Deprecate ``find_pylintrc``

DanielNoord opened the pull request:
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

I wonder if this is considered external. It is in the `__all__` but there is no reason for anybody to use this, I think? We don't even use it as the logic for finding the configuration file has moved to `find_default_config_files`.

Pierre-Sassoulas commented:
The coverage decreased because there were tests to check that we could browse a python package to find the pylintrc in parent directory too.

I would be in favor of searching at the root of the project in pyproject.toml, setup.cfg, pylintrc and .pylintrc and only that.  It's already a lot more than what flake8 is doing for example.  Guess which project has less issues to handle ? 😄  I think it's the way pylint is used most of the time (or should be used). And we added the possibility to do ``pylint .`` recently.But it's definitely a breaking change and I also think it's contrary to the per-directory configuration that was a big project of @AWhetter. 

This is related to https://github.com/PyCQA/pylint/pull/3769 where they were discussion about configuration.

DanielNoord commented:
> The coverage decreased because there were tests to check that we could browse a python package to find the pylintrc in parent directory too.
> 
> I would be in favor of searching at the root of the project in pyproject.toml, setup.cfg, pylintrc and .pylintrc and only that. It's already a lot more than what flake8 is doing for example. Guess which project has less issues to handle ? 😄 I think it's the way pylint is used most of the time (or should be used). And we added the possibility to do `pylint .` recently.But it's definitely a breaking change and I also think it's contrary to the per-directory configuration that was a big project of @AWhetter.
> 
> This is related to #3769 where they were discussion about configuration.

The issue is that these tests don't test that.. I removed these tests because the only asserts are on `find_pylintrc` which we don't use internally.
I have a refactor of `find_default_config_files` ready to go, so I would be more than happy to add similar tests for that function in a follow-up PR.

Pierre-Sassoulas reviewed (changes requested):
I think we should deprecate ``find_pylintrc`` first, before removing in 3.0, it's an API. We can add the new tests in this MR too.

Pierre-Sassoulas reviewed (approved):
Lgtm, thank you.
